components/camel-yql/src/main/java/org/apache/camel/component/yql/client/YqlClient.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.yql.client;
+
+import java.net.URI;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class YqlClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(YqlClient.class);
+    private static final CloseableHttpClient HTTP_CLIENT = HttpClients.createDefault();
```

Comment:
Can we avoid a static shared client in the JVM? This dont work well on app servers etc. Is the HTTP_CLIENT thread safe also? If it should be shared/reused then we can have a http client on the component that they producers can reuse.

Reply:
Yes `CloseableHttpClient` is thread safe. I added a connection manager that can be set on the component.